src/store/moviesSlice.js

```diff
@@ -36,6 +64,22 @@ export const moviesSlice = createSlice({
       .addCase(fetchMovies.rejected, (state, action) => {
         state.error = action.error;
         state.status = 'failed';
+      })
+      // TODO: handle errors for add/edit/delete to show them on UI
```

Comment:
I think this can also be done just like how `fetchMovies.rejected` is handled by setting update `error` and `status` fields of the state:

```
addCase(addMovie.rejected, (state, action) => {
        state.error = action.error;
        state.status = 'failed';
})
```

Then it can be shown on the UI with already existing `ErrorBoundary` component

Reply:
Added new handlers. In MovieList.js there is a check for `failed` status and the simple `alert()` to show error messages.
<img width="486" alt="image" src="https://github.com/f8B23Lef/reactJS-course-project/assets/25364800/329a3298-9dee-49bb-b476-c5bc315baf3b">
In case of error on Add/Edit/Delete movie action all `content` disappeared from UI, but I suppose it shouldn't. How we can manage this?
I tried to still show `content` like for `succeeded` status.  But in this case, every time you click on smth (after the 1st error appears) alert is displayed. How can we get rid of this? Or what the common approach for such cases?

Reply:
Regarding the content disappearing from the UI - the `content` is only being set when the `moviesStatus` is `succeeded`. But when the status is `failed`, the `content`'s value is null and that's why there is nothing being rendered on the UI when there is an error. I've made some changes to the component to make it work so that the content will always be there regardless of `failed` or `succeeded` status (this is just a quick solution, please feel free to come up with your own one):

![image](https://github.com/f8B23Lef/reactJS-course-project/assets/60813978/5e0de5e5-e1b0-471b-9546-f4f35bcabf2c)


About the case where the alert is being shown after each click, it is because the flag `shouldRefetchMovies` is still set to `true`  (or hasn't been changed) in some of the `rejected`, `fulfilled` cases in `moviesSlice` which is leading to unwanted refetchs and alerts. Please set the `shouldRefetchMovies` flag to false when you don't want to refetch data.

Reply:
Done.